keel-plugin/src/main/kotlin/com/netflix/spinnaker/keel/plugin/ResourceHandler.kt

```diff
@@ -83,6 +83,17 @@ interface ResourceHandler<T : Any> : KeelPlugin {
 }
 
 /**
- * Represents some kind of conflict that prevents the plugin from interacting with the resource.
+ * Searches a list of `ResourceHandler`s and returns the first that supports [apiVersion] and
+ * [kind].
+ *
+ * @throws UnsupportedKind if no appropriate handlers are found in the list.
  */
-class ResourceConflict(message: String) : RuntimeException(message)
+internal fun List<ResourceHandler<*>>.supporting(
```

Comment:
We were doing this in a couple of different places so I made an extension function here that can just get the handler for an `apiVersion`/`kind` from a list of handlers.